src/tests/multihost/alltests/test_ns_account_lock.py

```diff
@@ -0,0 +1,340 @@
+        ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+        user_dn = 'uid=foo3,ou=People,dc=example,dc=test'
+        role_dn = "filtered"
+        add_member = [(ldap.MOD_DELETE, 'o', role_dn.encode('utf-8'))]
+        (ret, _) = ldap_inst.modify_ldap(user_dn, add_member)
+        assert ret == 'Success'
+        ssh1 = SSHClient(client_e,
+                         username="foo3@example1",
+                         password="Secret123")
+        ssh1.close()
+        time.sleep(3)
+        unlock_check(multihost, "foo3")
+        # Activate filtered role
+        clean_sys(multihost)
+        manage_user_roles(multihost, "cn=filtered", "unlock", "role")
+        ssh1 = SSHClient(client_e,
+                         username="foo4@example1",
+                         password="Secret123")
+        ssh1.close()
+        time.sleep(3)
+        unlock_check(multihost, "foo4")
+
+    def test_nested_role_inactivated(self, multihost):
+        """
+        title: Nested role has both the above roles and inactivated
+        :id: 312e42c8-3045-11ec-88d4-845cf3eff344
+        :steps:
+            1. Add nasted role and make it inactive
+            2. Nested role has the managed role
+            3. Nested role has the filtered role
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        clean_sys(multihost)
+        client_e = multihost.client[0].ip
+        master_e = multihost.master[0].ip
+        ldap_uri = f'ldap://{master_e}'
+        ds_rootdn = 'cn=Directory Manager'
+        ds_rootpw = 'Secret123'
+        ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+        user_info = {'cn': 'nested'.encode('utf-8'),
+                     'objectClass': [b'top',
+                                     b'LdapSubEntry',
+                                     b'nsRoleDefinition',
+                                     b'nsComplexRoleDefinition',
+                                     b'nsNestedRoleDefinition'],
+                     'nsRoleDN': [b'cn=filtered,ou=people,'
+                                  b'dc=example,dc=test',
+                                  b'cn=managed,ou=people,'
+                                  b'dc=example,dc=test']}
+        user_dn = 'cn=nested,ou=People,dc=example,dc=test'
+        (_, _) = ldap_inst.add_entry(user_info, user_dn)
+        manage_user_roles(multihost, "cn=nested", "lock", "role")
+        with pytest.raises(paramiko.ssh_exception.AuthenticationException):
+            SSHClient(client_e,
+                      username="foo1@example1",
+                      password="Secret123")
+        time.sleep(3)
+        lock_check(multihost, "foo1")
+        with pytest.raises(paramiko.ssh_exception.AuthenticationException):
+            SSHClient(client_e,
+                      username="foo4@example1",
+                      password="Secret123")
+        time.sleep(3)
+        lock_check(multihost, "foo4")
```

Comment:
User foo1 is added to managed role in test_inactive_managed_roles and foo4 in the filter_role test. Maybe the addtion of these users should also be part of the fixture.